### PR TITLE
feat: modify init command to not use placeholders

### LINKED ISSUE #239: Modify charmil init command to replace text using regex instead of placeholders
Related to issue #240 
## Description
The `charmil init` command uses templates and placeholders present in the [charmil-starter](https://github.com/aerogear/charmil-starter) repo to generate a starter project, which makes the `charmil-starter` repo unusable without the init command (ie. users cannot clone and use it directly).

As discussed earlier, we need the charmil init command to use regex for replacing the necessary text fields instead of using templates and placeholders.

### PR DESCRIPTION
**Pull request by namit-chandwani**
Closes #239 

# Description

- The `init` command of Charmil CLI uses the template files and their placeholders that are currently present in the `charmil-starter` repo to generate a starter project, which makes the `charmil-starter` repo unusable without the init command (ie. users cannot clone the repo and use it directly due to the presence of template files).

- Our aim is to provide an additional option for users to use Charmil Starter by allowing them to directly clone the [charmil-starter](https://github.com/aerogear/charmil-starter) Github repository.

- This can be achieved by following these 2 steps:
   - Change the init command implementation to replace default values with the user-specified ones (ie. Github org/user name, repo name and CLI name) by performing simple String replacements instead of using go templates and substituting its placeholders.
     _**[Handled by this PR]**_

   - Remove the current contents of the [charmil-starter](https://github.com/aerogear/charmil-starter) repo (as it contains template files instead of runnable code) and replace it with the contents of the starter present in this repo ([charmil/starter](https://github.com/aerogear/charmil/tree/main/starter)). 
   **_[Handled by the PR: aerogear/charmil-starter/pull/7]_**

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

# Checklist

- [x] Documentation added for the feature
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by namit-chandwani** [comment]
This PR is now ready for review :)

CC: @wtrocki

**Comment by wtrocki** [comment]
ETA 1 day. Weekend

File: starter/Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ BUILDFLAGS := -gcflags "all=-N -l" $(BUILDFLAGS)
 endif
 
 # The details of the application:
-binary:=cli
+binary:=startercli
```
Comment (wtrocki) [inline starter/Makefile:15]: Lets put somethig very unique as name and meaningfull.

startervarname placeholdercli etc.

StarterCli does generate false positives

File: starter/cmd/startercli/locales/en/completion.en.yaml [hunk 2]
```diff
@@ -31,11 +31,11 @@ completion.bash.cmd.longDescription:
   one: '''
   Generate command completion script for Bash shell.
   Installing on Linux:
-  1. Run `cli completion bash > cli_completions` to create the script file.
-  2. Run `sudo mv cli_completions /etc/bash_completion.d/cli` to place the script in a special Bash completions directory.
+  1. Run `startercli completion bash > startercli_completions` to create the script file.
```
Comment (wtrocki) [inline starter/cmd/startercli/locales/en/completion.en.yaml:34]: Maybe ignore cli name here.
Also CLI name can be build var

**Review by wtrocki — APPROVED** [review]
good stuff but left some suggestions

**Comment by wtrocki** [comment]
I do not think all uses of cli name were properly replaced :)

**Comment by namit-chandwani** [comment]
> good stuff but left some suggestions

Sorry, missed the notification somehow.
Will make the required changes tomorrow :)

File: starter/Makefile (hunk 1 above)
Comment (namit-chandwani) [reply starter/Makefile]: Resolved via commit: https://github.com/aerogear/charmil/pull/243/commits/d9daf62f3f2be1b9de2af2fc23a9a0814258e776

File: starter/cmd/startercli/locales/en/completion.en.yaml (hunk 2 above)
Comment (namit-chandwani) [reply starter/cmd/startercli/locales/en/completion.en.yaml]: Resolved via commit: https://github.com/aerogear/charmil/pull/243/commits/d9daf62f3f2be1b9de2af2fc23a9a0814258e776

**Comment by namit-chandwani** [comment]
Merging this PR now and making a new Charmil release for both `charmil/starter` module and `charmil-starter` repo to use.

**Pull request merged**